### DISCUSSION TITLE
Can I initialize new orm instance every time rather than em.fork() or RequestContext helper?

### DISCUSSION THREAD

**Discussion by suerta-git** [mikro-orm/mikro-orm#2613] [Q&A]
Hi there, I am building my serverless application and don't use any DI container.

As the title I said, in this case can I initialize new orm instance (`MikroORM.init(config)`) every time my lambda triggered?

And what's the pros and cons comparing with keeping single instance then forking entity manager?

**Comment by B4nan** [comment]
Forking is fast, initializing the ORM is not. Forking is just about creating some class instances, all resources are shared.

I guess in the context of serverless app it is fine to do this, but never did any serverless app myself so can't help much with the best practices. This example app we have apparently does the same.

https://github.com/thomaschaaf/serverless-mikro-orm-example-app/blob/master/src/handlers/author.ts#L9

One thing you could consider to speed up the init phase could be to use webpack, I remember some users are doing that. But it can be challenging to setup webpack with MikroORM, it will bring some additional limitations. Can't really help with that either, not a webpack user unless I am forced to :P We have this page in the docs, provided by the community: 

https://mikro-orm.io/docs/deployment/#deploy-a-bundle-of-entities-and-dependencies-with-webpack

**Reply by suerta-git** [reply]
Thanks a lot. Seems webpack is too complex for my use case, so I will adopt the forking way simply.

But what about forking vs request context helper? Can you please explian the differences between them?

**Reply by B4nan** [reply]
`RequestContext` is just about using DI containers. It also forks in the background, it works the same way. Only difference is that inside the RequestContext handler, you can use the global EM or repository instances, while they will automatically use the contextual fork.
